libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -571,23 +540,53 @@ private Optional<JsonNode> generateAddressClaim(List<VerifiableCredential> vcs)
             return Optional.empty();
         }
 
-        var addressNode =
-                extractSubjectDetailFromVc(
-                        ADDRESS_PROPERTY_NAME,
-                        addressVc.get(),
-                        "Error while parsing Address CRI credential",
-                        ErrorResponse.FAILED_TO_GENERATE_ADDRESS_CLAIM);
+        var credentialSubject =
+                ((IdentityCheckCredential) addressVc.get().getCredential()).getCredentialSubject();
```

Comment:
The address VC should be an `AddressCredential`, not `IdentityCheckCredential`!

Also, it's a bit safer to do this with a checked cast, e.g.:
```
if (addressVc.get().getCredential() instanceof AddressCredential addressCredential) {
    var address = addressCredential.getCredentialSubject().getAddress();
    ...
} else {
    // error - not an address credential!
}
```